src/server/room-controller.js

```diff
@@ -118,9 +122,13 @@ export default class RoomController {
 
   async handleGetRoomPage(req, res) {
     const { roomId } = req.params;
-    const { _id: userId } = req.user;
+    const userId = req.user?._id;
     const routeWildcardValue = urls.removeLeadingSlash(req.params['0']);
 
+    if (!userId) {
```

Comment:
This should also never happen as this page is only server after the `needsPermission(permissions.OWN_ROOMS)` middleware.

Here, I think it is also the case that we cannot use this middleware as we have to allow unauthenticated users to access public rooms.

Reply:
Please see comment here 👉 https://github.com/educandu/educandu/pull/189#discussion_r793389400